persistence-api/src/main/java/io/stargate/db/Persistence.java

```diff
@@ -94,6 +94,11 @@ default boolean supportsSecondaryIndex() {
   /** Returns true if the persistence backend supports Storage Attached Indexes. */
   boolean supportsSAI();
 
+  default boolean useLoggedBatches() {
+    return Boolean.parseBoolean(
+        System.getProperty("stargate.persistence.loggedbatches.support.default", "true"));
```

Comment:
If we delegate this decision to `Persistence` why use a system property here? I guess it should be done in `Persistence` implementations so that the returned value were provided thoughtfully.

Reply:
Conversely, if we want to control Docs API behaviour independently of the persistence impl, shouldn't the property be defined somewhere in Doc API classes instead?

Reply:
I guess if it's not always explicitly defined in the Persistence implementation, then sometimes you would want to configure it by setting properties at runtime? To be honest I just did it in an analogous way to how supports2i was done and didn't think much of it

Reply:
Yeah... I know. I guess is those earlier days we wanted to keep internal interface backward-compatibility as much as possible :thinking: now, though, I think we have a lot of change going on at the persistence level, that keeping strict compatibility is no longer practical.

Making the new API mandatory, will draw attention from `Persistence` implementors, which is a good thing in this situation, I think.

@dougwettlaufer : what is you take on this?

Reply:
I think this behavior is pretty closely tied to the persistence though - maybe it shouldn't be exposed as a property at all? (exposing the property does make it a bit more straightforward to test though)

Reply:
For testing, would it be possible to allow overrides in Docs API code? As in: if override is not set, use `Persistence.supportsLoggedBatches` and let tests override?

Reply:
Yea that sounds more right - I'll change to that

Reply:
@dimas-b I can see it both ways. I get that keeping consistent with the 2i makes sense but personally I'd lean to deferring this to the services to decide. As in maybe docsapi wants to use logged batches but graphql is fine with unlogged. But really I'm good either way.

Reply:
yeah, I think we have it this way in the latest commit. `Persistence` declares whether logged batches are supported and Docs API has a `-D` that controls whether they are actually used or not (defaults to the persistence-level flag).